internal/controller/container_reconciler.go

```diff
@@ -26,8 +27,18 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/log"
 )
 
+const latestUploadPath = "uploads/latest.tar.gz"
+
 type ContainerizedObject interface {
-	object
+	client.Object
+
+	GetConditions() *[]metav1.Condition
+
+	SetStatusReady(bool)
+
+	GetStatusImage() apiv1.ImageStatus
+	SetStatusImage(apiv1.ImageStatus)
```

Comment:
nit: Seeing that you defined `GetArtifactStatus` within types I would expect the getters and setters to be consistent

Reply:
You mean `GetArtifactStatus` should be renamed to `GetStatusArtifacts`? I agree with that.

Reply:
Yeah you got my drift 🙏